docs/MIGRATION-GUIDE.md

```diff
@@ -0,0 +1,208 @@
+# Migrating to webpack-blocks v1.0
+
+## Table of Contents
+
+<!-- To update run: npx markdown-toc --maxdepth 2 -i docs/MIGRATION-GUIDE.md -->
+
+<!-- toc -->
+
+- [Migrating configuration](#migrating-configuration)
+- [Additional features](#additional-features)
+- [Migrating blocks](#migrating-blocks)
+
+<!-- tocstop -->
+
+
+## Migrating configuration
+
+### `match()`
+
+You can now use `match()` to specify on which files to apply certain loaders. Works with every block that adds a loader, like `babel`, `css`, `elm`, `postcss`, `sass`, ...
+
+The blocks still work without `match()`. **However, blocks dropped the `fileType`, `exclude` & `include` from their options. Use `match()` for that.**
+
+```js
+const { createConfig, css, file, match, postcss, url } = require('webpack-blocks')
+const path = require('path')
+
+module.exports = createConfig([
+  babel(),            // matches *.js, *.jsx outside node_modules/ by default
+  match('*.css', { exclude: path.resolve('node_modules') }, [
```

Comment:
Weird comment formatting, also why use `path.resolve` when a regex `/node_modules/` works fine? The main readme uses `path.resolve` as well...

Reply:
Yeah, I just copied the snippet. Where would you put the comment?

`path.resolve()` vs regex seems almost like a philosophical question. One is precise, but longer, the other one is concise, but inprecise, yet will do the job as well.

Reply:
I'd put it right next to the block, and would also add new lines between blocks for better readability.

Reply:
Isn't it next to the block? (Maybe you can just paste an example of what you suggest)

Reply:
There is a huge whitespace between
![screen shot 2017-09-22 at 13 42 39](https://user-images.githubusercontent.com/7415966/30741035-fce31cd4-9f9b-11e7-9f4c-3caca0a58099.png)

Reply:
Now we're talking. How about this?

```js
const { createConfig, css, file, match, postcss, url } = require('webpack-blocks')
const path = require('path')

module.exports = createConfig([
  babel(),    // matches *.js, *.jsx outside node_modules/ by default

  match('*.css', { exclude: path.resolve('node_modules') }, [
    css(),
    postcss()
  ]),

  match(['*.gif', '*.jpg', '*.jpeg', '*.png', '*.svg', '*.webp'], [
    file()
  ])
])
```

Reply:
I’d put it on top of the block, and capitalize the first letter.